rust/osm2lanes/src/lib.rs

```diff
@@ -124,50 +101,70 @@ pub trait LanePrintable {
     fn as_utf8(&self) -> char;
 }
 
-impl LanePrintable for LaneType {
+impl LanePrintable for Lane {
     fn as_ascii(&self) -> char {
         match self {
-            LaneType::Driving => 'd',
-            LaneType::Biking => 'b',
-            LaneType::Bus => 'B',
-            LaneType::Parking => 'p',
-            LaneType::Sidewalk => 's',
-            LaneType::Shoulder => 'S',
-            LaneType::SharedLeftTurn => 'C',
-            LaneType::Construction => 'x',
-            LaneType::Buffer(_) => '|',
+            Self::Travel {
+                designated: LaneDesignated::Foot,
+                ..
+            } => 's',
+            Self::Travel {
+                designated: LaneDesignated::Bicycle,
+                ..
+            } => 'b',
+            Self::Travel {
```

Comment:
Trade-off: This kind of code becomes more complex, having to match something embedded. But it buys us flexibility.